### PR TITLE
Added helper functions (see #137)

### PR DESCRIPTION
**Pull request by mutt0-ds**
Hello! I've worked on the issue #137 and added the helper functions following the same structure of notion-sdk-js.

I have a couple of doubts about the typing in the Async version, since I'm rarely working with this kind of functions, so please let me know if it's ok.

I tested them locally with my Notion. but I couldn't add tests because they require a mock call to the API... Should we raise an issue and work on it?

We also need to update the README, I'll do it if the request is approved

### CODE REVIEW

File: notion_client/helpers.py [hunk 1]
```diff
@@ -24,3 +24,78 @@ def get_id(url: str) -> str:
         raise ValueError("The path in the URL seems to be incorrect.")
     raw_id = path[-32:]
     return str(UUID(raw_id))
+
+
+def iterate_paginated_API(
+    function: Callable[..., Any], first_page_args: Dict[Any, Any]
```
Comment (ramnes) [inline notion_client/helpers.py]: Let's use kwargs?

```suggestion
    function: Callable[..., Any], **kwargs: Any
```

File: notion_client/helpers.py [hunk 2]
```diff
@@ -24,3 +24,78 @@ def get_id(url: str) -> str:
         raise ValueError("The path in the URL seems to be incorrect.")
     raw_id = path[-32:]
     return str(UUID(raw_id))
+
+
+def iterate_paginated_API(
```
Comment (ramnes) [inline notion_client/helpers.py]: IIRC Python functions must always be lowercase

```suggestion
def iterate_paginated_api(
```

File: notion_client/helpers.py [hunk 3]
```diff
@@ -24,3 +24,78 @@ def get_id(url: str) -> str:
...
+) -> Generator[List[Any], None, None]:
+    """Return an iterator over the results of any paginated Notion API."""
+    next_cursor = None
+
+    while True:
+        response = function(**first_page_args, start_cursor=next_cursor)
+        yield response.get("results")
+
+        next_cursor = response.get("next_cursor")
+        if (not response["has_more"]) | (next_cursor is None):
```
Comment (ramnes) [inline notion_client/helpers.py]:
```suggestion
        if not response.get("has_more") or not next_cursor:
```

**Review by ramnes — CHANGES_REQUESTED** [review]
LGTM overall, just a few little things to change

**Comment by ramnes** [comment]
Thank you very much for taking this work!

I would love if the README changes could be added to the same commit as well. :)

For the tests, it's true that cassettes are great for me as a sole-developer but not practical at all for collaboration. Do you have any idea on how we could do things differently? c.f. https://github.com/ramnes/notion-sdk-py/pull/116#issuecomment-1101463557

**Comment by mutt0-ds** [comment]
I'll work on the README tomorrow, while I'm thinking about an easy way for handling tests. We can mockup all the responses or maybe use a secret token associated to the notion example at [test_helpers](https://github.com/ramnes/notion-sdk-py/blob/main/tests/test_helpers.py)?

**Comment by ramnes** [comment]
I think the key is to not rely on existing pages or other data we can't control. We should just give the tests a private token and a root page, and then they'd bootstrap a test environment (i.e. a few pages) within the given root page.

That way, anyone could run the tests with their own Notion account. We could still use the cassettes to speed up most tests, but that wouldn't be a blocker anymore to add more tests.

File: notion_client/helpers.py (hunk 2 above)
Comment (ramnes) [reply notion_client/helpers.py]: Did you resolve it but forgot to include the changes in the commit? :sweat_smile:

File: notion_client/helpers.py [hunk 4]
```diff
@@ -24,3 +24,76 @@ def get_id(url: str) -> str:
...
+
+
+def iterate_paginated_API(
+    function: Callable[..., Any], **kwargs: Any
+) -> Generator[List[Any], None, None]:
+    """Return an iterator over the results of any paginated Notion API."""
+    next_cursor = None
+
+    while True:
+        response = function(kwargs, start_cursor=next_cursor)
```
Comment (evertheylen) [inline notion_client/helpers.py]: Is this not supposed to be `**kwargs`? Awaiting the merge of this PR, I copied the code but I had to change this to make it work.

**Comment by mutt0-ds** [comment]
@ramnes OK about tests, I agree. 
I'll think about the setup in the next days (and probably open a new issue or referring to the existing one about coverage). Thanks!

**Comment by davideph** [comment]
> LGTM overall, just a few little things to change

Changes done!

File: notion_client/helpers.py (hunk 3 above)
Comment (ramnes) [reply notion_client/helpers.py]: Did you forget to commit these changes?

**Comment by ramnes** [comment]
One little thing left :)

**Comment by ramnes** [comment]
Thanks! 🚀

**Pull request merged**